azurerm/internal/services/batch/resource_arm_batch_pool.go

```diff
@@ -583,6 +594,13 @@ func resourceArmBatchPoolUpdate(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	if d.HasChange("metadata") {
```

Comment:
this is a create only function so i don't think we need to check this and can just always set it?

Reply:
The _d.HasChange_ is in the _resourceArmBatchPoolUpdate_ method. Is this not used for updates?